File: Programming Basics with JS/T06 - While Loop/lecture/06. Max Number.js
function maxNumber(input) {
    let max = Number(input[0]);
    let i = 1;
    let currentNumber;

    while (input[i] !== 'Stop') {
        currentNumber = Number(input[i]);
        if (currentNumber > max) {
            max = currentNumber;
        }
        i++;
    }
    console.log(max);
}
maxNumber(["100", "99", "80", "70", "Stop"]);
maxNumber(["-10", "20", "-30", "Stop"]);
maxNumber(["45", "-20", "7", "99", "Stop"]);

File: Programming Basics with Go/T06 - Advanced Loops/lecture/06. MaxNumber.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func main() {

	var min int = math.MaxInt32 //минималното число
	//повтаряме: число  -> проверка дали е максимум
	//спираме: входни данни == "Stop"
	//продължаваме: входни данни != "Stop"
	//входни данни (текст): "Stop" или число под формата на текст ("199")
	var input string
	fmt.Scanln(&input)

	for input != "Stop" {
		//input e число под формата на текст ("199")
		//1. input = "199" -> цяло число = 199
		//преобразувание от текст към число
		number, _ := strconv.Atoi(input)
		//2. проверка дали цяло число е мин
		if number < min {
			min = number
		}

		fmt.Scanln(&input)
	}

	fmt.Print(min)

}

